src/App/Tracker/Controller/Issue/Save.php

```diff
@@ -333,4 +339,133 @@ private function updateGitHub($issueNumber, array $data, $state, $oldState)
 
 		return $this;
 	}
+
+	/**
+	 * Add the status as label on GitHub.
+	 *
+	 * The method gets the list of current labels and remove all labels that starts with `~`
+	 * as it is the status prefix, after this the method add the new status label
+	 *
+	 * @param   integer  $issueNumber  The issue number.
+	 * @param   string   $newlabel     The new label that should be added as label.
+	 *
+	 * @throws \Exception
+	 * @throws \JTracker\Github\Exception\GithubException
+	 *
+	 *
+	 * @return  $this
+	 *
+	 * @since   1.0
+	 */
+	private function addLabelToGitHub($issueNumber, $newlabel)
+	{
+		/* @type \JTracker\Application $application */
+		$application = $this->getContainer()->get('app');
+
+		// Get the labels for the pull's issue
+		try
+		{
+			$labels = $github->issues->get($project->gh_user, $project->gh_project, $issueNumber)->labels;
+		}
+		catch (\DomainException $e)
+		{
+			$application->enqueueMessage(
+				sprintf(
+					'Error retrieving labels for GitHub item %s/%s #%d - %s',
+					$project->gh_user,
+					$project->gh_project,
+					$issueNumber,
+					$e->getMessage()
+				),
+				'error'
+			);
+
+			return;
+		}
+
+		// Check if we have any label
+		if (count($labels) > 0)
+		{
+			foreach ($labels as $label)
+			{
+				if ($label->name == $new_label)
+				{
+					$LabelIsSet = true;
+				}
+
+				$prefix = substr($label->name, 0, 1);
+
+				if ($prefix == '~')
```

Comment:
If changed - yes. But currently you remove also the label in case when it is not changed. So when `$LabelIsSet = true;` you should not remove the lable.